python/hail/linalg/blockmatrix.py

```diff
@@ -51,23 +51,100 @@ def read(cls, path):
         return cls(Env.hail().linalg.BlockMatrix.read(
             hc._jhc, path))
 
+    @classmethod
+    @typecheck_method(uri=str,
+                      n_rows=int,
+                      n_cols=int,
+                      block_size=nullable(int))
+    def fromfile(cls, uri, n_rows, n_cols, block_size=None):
+        """Creates a block matrix from a binary file.
+
+        Examples
+        --------
+
+        >>> import numpy as np
+        >>> a = np.random.rand(10, 20)
+        >>> a.tofile('/local/file') # doctest: +SKIP
+
+        To create a block matrix of the same dimensions:
+
+        >>> from hail.linalg import BlockMatrix
+        >>> bm = BlockMatrix.fromfile('file:///local/file', 10, 20) # doctest: +SKIP
+
+        See also :meth:`BlockMatrix.from_numpy`.
+
+        Notes
+        -----
+        This method, analogous to `numpy.fromfile
+        <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.fromfile.html>`__,
+        reads a binary file of float64 values in row-major order, such as that
+        produced by `numpy.tofile
+        <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.fromfile.html>`__
+        or :meth:`BlockMatrix.tofile`.
+
+        The number of entries must be less than :math:`2^{31}`.
+
+        Use :meth:`BlockMatrix.write` and :meth:`BlockMatrix.read` to save and
+        restore block matrices.
```

Comment:
save and load.  Also, you might say in the case that you don't need to interoperate with numpy, in particular, because read/write read and write in parallel.